### PR TITLE
Deny bucket listing in iam policy

### PR DESCRIPTION
**Pull request by Kidswiss**
Previously it was possible that an IAM key could list all buckets in the project, even the ones where the IAM key didn't have any other permissions.

We now explicitly deny the listing of buckets. Also, already existing IAM keys will be updated with the new policy.

## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] I have run successfully `make test-e2e` locally.
- [x] Update the documentation.
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [x] Link this PR to related issues or PRs.

### CODE REVIEW

File: test/e2e/kafka/00-assert.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ spec:
     size:
       plan: startup-2
     zone: ch-dk-2
-    version: '3.2'
+    version: '3.5'
```
Comment (wejdross) [inline test/e2e/kafka/00-assert.yaml:24]: is it anyhow related?

**Review by wejdross — APPROVED** [review]
LGTM

File: operator/iamkeycontroller/update.go [hunk 2]
```diff
@@ -5,13 +5,22 @@ import (
...
 func (p *IAMKeyPipeline) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
 	log := controllerruntime.LoggerFrom(ctx)
-	log.V(1).Info("Updating resource (noop)")
-	return managed.ExternalUpdate{}, nil
+	log.V(1).Info("Updating role")
+
+	iamKey := fromManaged(mg)
+	iamKey.SetConditions(exov1.Updating())
+
+	role := createRole(iamKey.Spec.ForProvider.KeyName, iamKey.Spec.ForProvider.Services.SOS.Buckets)
```
Comment (zugao) [inline operator/iamkeycontroller/update.go:21]: Why do you create the role in `Update()` method?

**Review by zugao — APPROVED** [review]
LGTM, I haven't tested it. You might want to test it throughly as I don't see any unit/e2e tests.

**Comment by Kidswiss** [comment]
> LGTM, I haven't tested it. You might want to test it throughly as I don't see any unit/e2e tests.

I've tested it via kind.

I agree about the unit/e2e test, but on the other hand this logic should actually be part of a composition instead of the provider, as it's very specific to our use-case.

File: operator/iamkeycontroller/update.go (hunk 2 above)
Comment (Kidswiss) [reply operator/iamkeycontroller/update.go:21]: `createRole()` creates the role object with the policy, it doesn't create anything on the API.

Should I rename this function?

File: test/e2e/kafka/00-assert.yaml (hunk 1 above)
Comment (Kidswiss) [reply test/e2e/kafka/00-assert.yaml:24]: Somewhat, the e2e-test doesn't work anymore with 3.2.

**Pull request merged**